src/main/kotlin/note/application/NoteUpdater.kt

```diff
@@ -12,6 +12,14 @@ class NoteUpdater(
     private val repository: NoteRepository,
     private val identifierGenerator: NoteIdentifierGenerator,
 ) {
+
+    /***
+     * Updates a Note with the given identifier, with the given title and description,
+     * @see Note
+     * @throws NonExistentNoteException if the note does not exist
+     * @throws UnchangedNoteException if the note is unchanged
+     * @return The newly created Note, with the corresponding updates
+     */
     fun update(identifier: NoteIdentifier, newTitle: String, newDescription: String?): Note {
         val note = repository.search(identifier) ?: throw NonExistentNoteException(identifier)
         if (unchangedNote(note, newTitle, newDescription)) throw UnchangedNoteException(identifier)
```

Comment:
This should receive a string primitive instead of a NoteIdentifier instance, also note that there's no need to create a class to represent a note identifier, Identifier class can be used instead.

Reply:
That's not code from my PR, it's instead code that was already implemented. I was just adding the needed docs for it.
So a new PR should be issued with those changes, instead of requesting them in this one, which has nothing to do with the docs I've added.
I'll do the changes later though.

Reply:
I was fixing this on the refactor-identifier branch